snapd-glib/snapd-client.c

```diff
@@ -1337,6 +1338,54 @@ snapd_client_get_auth_data (SnapdClient *self)
     return priv->auth_data;
 }
 
+
+/**
+ * snapd_client_get_notices_async:
+ * @client: a #SnapdClient.
+ * @cancellable: (allow-none): a #GCancellable or %NULL.
+ * @callback: (scope async): a #GAsyncReadyCallback to call when the request is satisfied.
+ * @user_data: (closure): the data to pass to callback function.
+ *
+ * Asynchronously get notifications that have occurred / are occurring on the snap daemon.
+ *
+ * Since: 1.29
+ */
+void
+snapd_client_get_notices_async (SnapdClient *self,
+                                GCancellable *cancellable, GAsyncReadyCallback callback, gpointer user_data)
+{
+    g_return_if_fail (SNAPD_IS_CLIENT (self));
+
+    g_autoptr(SnapdGetNotices) request = _snapd_get_notices_new (NULL, NULL, NULL, NULL, 0, cancellable, callback, user_data);
+    send_request (self, SNAPD_REQUEST (request));
+}
+
+/**
+ * snapd_client_get_notices_finish:
+ * @client: a #SnapdClient.
+ * @result: a #GAsyncResult.
+ * @error: (allow-none): #GError location to store the error occurring, or %NULL to ignore.
+ *
+ * Complete request started with snapd_client_get_notices_async().
+ *
+ * Returns: (transfer full) (element-type GSList): a list of #SnapdNotice or %NULL on error.
+ *
+ * Since: 1.29
+ */
+GSList *
+snapd_client_get_notices_finish (SnapdClient *self, GAsyncResult *result, GError **error)
+{
+    g_return_val_if_fail (SNAPD_IS_CLIENT (self), NULL);
+    g_return_val_if_fail (SNAPD_IS_GET_NOTICES (result), NULL);
+
+    SnapdGetNotices *request = SNAPD_GET_NOTICES (result);
+
+    if (!_snapd_request_propagate_error (SNAPD_REQUEST (request), error))
+        return NULL;
+    return _snapd_get_notices_get_notices (request);
```

Comment:
This highlights another reason this should be a GPtrArray - if the list was empty this would return NULL which would look like an error.

Reply:
You are right.